### ISSUE TITLE
[php] Null assignment to a variable holding callback function get's lost somehow

### ISSUE THREAD

**Issue by sebthom** [HaxeFoundation/haxe#5427]
When executing the following code, the ClassB instance ends up suddenly having a non-null value for the comparator argument, despite the fact that ClassA is instantiated with a null value for the comparator which is passed to the ClassB instance.

```haxe
class ClassB {   
    public function new(comparator:String -> String -> Int) {
        trace('ClassB: comparator is null: ' + (comparator == null));
    }
}

class ClassA {
    var cmp:String -> String -> Int = null;
    
    public function new(comparator:String -> String -> Int) {
        trace('ClassA: comparator is null: ' + (comparator == null));
        this.cmp = comparator;
        trace('ClassA: this.cmp is null: ' + (this.cmp == null));
    }
    
    public function createClassB() {
        return new ClassB(this.cmp);
    }
}

class Test {
    static function main() {
        new ClassA(null).createClassB();
    }
}
```

Output on PHP:
```
Test.hx:11: ClassA: comparator is null: true
Test.hx:13: ClassA: this.cmp is null: true
Test.hx:3: ClassB: comparator is null: false
```
Output on other targets:
```
Test.hx:11: ClassA: comparator is null: true
Test.hx:13: ClassA: this.cmp is null: true
Test.hx:3: ClassB: comparator is null: true
```

**Comment by ncannasse** [comment]
Might be a unwanted closure creation here.

**Comment by RealyUniqueName** [comment]
This is an old bug in php target: https://github.com/HaxeFoundation/haxe/issues/3650

**Comment by mockey** [comment]
I just hacked in my old idea from February 2015 (#3650) and it seems to work as far as I can tell.

**Comment by nadako** [comment]
Should be fixed by #5470
